include/lsst/afw/detection/Footprint.h

```diff
@@ -0,0 +1,383 @@
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+#if !defined(LSST_DETECTION_FOOTPRINT_H)
+#define LSST_DETECTION_FOOTPRINT_H
+
+#include <algorithm>
+#include <list>
+#include <set>
+#include <cmath>
+#include <cstdint>
+#include <memory>
+#include "ndarray.h"
+#include "lsst/base.h"
+#include "lsst/pex/policy/Policy.h"
+#include "lsst/afw/image/MaskedImage.h"
+#include "lsst/afw/image/Wcs.h"
+#include "lsst/afw/detection/Peak.h"
+#include "lsst/afw/geom/Span.h"
+#include "lsst/afw/geom/SpanSet.h"
+#include "lsst/afw/geom/Box.h"
+#include "lsst/afw/geom/ellipses.h"
+#include "lsst/afw/table/fwd.h"
+#include "lsst/afw/table/Schema.h"
+#include "lsst/afw/table/Key.h"
+#include "lsst/afw/table/io/Persistable.h"
+#include "lsst/afw/table/io/InputArchive.h"
+
+namespace lsst { namespace afw { namespace detection {
+
+/** @class lsst::afw::detection::Footprint
+ * @brief Class to describe the properties of a detected object from and image
+ *
+ * A Footprint is designed to be constructed with information about a detected
+ * object in an image. Internally a Footprint holds a SpanSet which is used to
+ * describe the location of the object in the image (the x, y pixel locations
+ * which are considered part of the object). In addition a Footprint contains
+ * a PeakCatalog which is used to store the location and intensity of peaks in
+ * the detection.
+ */
+class Footprint : public lsst::daf::base::Citizen,
+                  public afw::table::io::PersistableFacade<lsst::afw::detection::Footprint>,
+                  public afw::table::io::Persistable
+{
+public:
+    /*
+     * As a note there is no constructor which accepts a unique_ptr to a SpanSet.
+     * If someone did have a unique_ptr to a SpanSet it is possible to wrap the
+     * unique_ptr in a std::move when calling the constructor of a Footprint and
+     * the compiler will implicitly turn the unique_ptr into a shared_ptr and use
+     * that to construct the Footprint.
+     */
+    explicit Footprint(std::shared_ptr<geom::SpanSet> inputSpans,
+                       geom::Box2I const & region=geom::Box2I());
+    explicit Footprint(std::shared_ptr<geom::SpanSet> inputSpans,
+                       afw::table::Schema const & peakSchema,
+                       geom::Box2I const & region=geom::Box2I());
+    explicit Footprint(): lsst::daf::base::Citizen(typeid(this)),
+                          _spans(std::make_shared<geom::SpanSet>()),
+                          _peaks(PeakTable::makeMinimalSchema()),
+			              _region(geom::Box2I()) {}
```

Comment:
Add newlines between and docstrings for these.  Weird indentation on the last one.